Plutarch/Builtin.hs

```diff
@@ -19,6 +22,22 @@ instance POrd PBuiltinByteString where
   x £<= y = punsafeBuiltin PLC.LessThanEqualsByteString £ x £ y
   x £< y = punsafeBuiltin PLC.LessThanByteString £ x £ y
 
+-- | Errors that may arise while using 'preadByteStr'.
+data ByteStringReadError = UnevenLength | InvalidHexDigit Char
+  deriving stock (Eq, Ord, Read, Show)
+
+-- | Interpret a hex string as a PBuiltinByteString.
+preadByteStr :: String -> Either ByteStringReadError (Term s PBuiltinByteString)
```

Comment:
We probably want a better name. `phexByteStr`? Also, while it in general is good practice to use total functions, in this case none of the ways of constructing the AST support monadisms, hence I think it's better to have `HasCallStack =>` and make the function partial.

Reply:
Feel free to choose a name yourself, but it should be clear from the name how the string is parsed.

Reply:
Went with `phexByteStr`. And made it partial with as suggested.